File: src/test/resources/Fig4Test.kt
import kotlin.test.Test
import kotlin.test.assertEquals

class Fig4(var s: String) {
    fun greet(@argLabel("_") person: String, @argLabel("from") hometown: String): String {
        s = "Hello ${person}!\n"
        s += "Glad you could visit from ${hometown}."
        return s
    }
    fun main() {
        greet("Bill", "Cupertino")
    }
}

class Fig4Test {

    @Test
    fun testFig4() {
        val f = Fig4("")
        val expected = "Hello Bill!\nGlad you could visit from Cupertino."
        f.greet("Bill", "Cupertino")
        assertEquals(expected, f.s)
    }

    @Test
    fun testFig4GreetAbel() {
        val f = Fig4("")
        val expected = "Hello Abel!\nGlad you could visit from Austin."
        f.greet("Abel", "Austin")
        assertEquals(expected, f.s)
    }
}

File: swift_results/Fig4Test.swift
import XCTest
#if false
let Ꮻimports = ["kotlin.test.Test", "kotlin.test.assertEquals"]
#endif

class Fig4 {
    var s: String = ""
    func greet(_ person: String, from hometown: String) {
        s += "Hello \(person)!\n"
        s += "Glad you could visit from \(hometown).\n"
    }
}

class Fig4Test: XCTestCase {
    func testFig4() {
        let f = Fig4()
        let expected = "Hello Bill!\nGlad you could visit from Cupertino.\n"
        f.greet("Bill", from: "Cupertino")
        XCTAssertEqual(f.s, expected)
    }

    func testFig4GreetAbel() {
        let f = Fig4()
        let expected = "Hello Abel!\nGlad you could visit from Austin.\n"
        f.greet("Abel", from: "Austin")
        XCTAssertEqual(f.s, expected)
    }
}